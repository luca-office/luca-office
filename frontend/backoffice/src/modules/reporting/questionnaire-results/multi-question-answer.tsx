import {css} from "@emotion/react"
import * as React from "react"
import {Text} from "shared/components"
import {FreetextAnswerFragment} from "shared/graphql/generated/FreetextAnswerFragment"
import {
  backgroundColorBright,
  backgroundMenuInactive,
  borderRadius,
  deepShadow,
  Flex,
  flex1,
  FontWeight,
  spacingSmall,
  TextSize
} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {FreeTextAnswerSwitch} from "./free-text-answer-switch/free-text-answer-switch"

export interface MultiQuestionAnswerProps {
  readonly percent: string
  readonly text?: string
  readonly selectionCount: number
  readonly freeTextAnswers?: FreetextAnswerFragment[]
  readonly participantCount: number
  readonly participantNames: Record<UUID, string>
}

export const MultiQuestionAnswer: React.FC<MultiQuestionAnswerProps> = props => {
  const {t} = useLucaTranslation()
  const {percent, text, selectionCount, freeTextAnswers, participantCount, participantNames} = props

  return (
    <div css={styles.container}>
      <div css={styles.percentage}>
        <Text size={TextSize.Medium} css={styles.label}>
          {percent}
        </Text>
      </div>
      <div css={styles.textContainer}>
        <Text size={TextSize.Medium} css={styles.label}>
          {t("common__x_of_y", {x: selectionCount, y: participantCount})}
        </Text>
        {freeTextAnswers !== undefined && freeTextAnswers.length > 0 ? (
          <FreeTextAnswerSwitch freeTextAnswers={freeTextAnswers} participantNames={participantNames} />
        ) : (
          <Text size={TextSize.Medium}>{text}</Text>
        )}
      </div>
    </div>
  )
}

const styles = {
  container: css(Flex.row, {
    backgroundColor: backgroundColorBright,
    boxShadow: deepShadow,
    borderRadius: borderRadius,
    boxSizing: "border-box"
  }),
  percentage: css({
    boxSizing: "border-box",
    width: 64,
    height: "100%",
    backgroundColor: backgroundMenuInactive,
    padding: spacingSmall
  }),
  textContainer: css({
    flex: flex1,
    padding: spacingSmall
  }),
  label: css({
    fontWeight: FontWeight.Bold,
    marginBottom: spacingSmall
  })
}
