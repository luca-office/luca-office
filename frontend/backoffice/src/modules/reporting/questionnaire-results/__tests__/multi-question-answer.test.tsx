import React from "react"
import {create} from "react-test-renderer"
import {MultiQuestionAnswer, MultiQuestionAnswerProps} from "../multi-question-answer"

const defaultProps: MultiQuestionAnswerProps = {
  selectionCount: 5,
  percent: "36 %",
  text: "test text",
  participantCount: 2,
  freeTextAnswers: [],
  participantNames: {}
}

describe("MultiQuestionAnswer", () => {
  it("renders correctly (SC/MC no FreeText)", () => {
    const component = create(<MultiQuestionAnswer {...defaultProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders correctly (SC/MC with FreeText)", () => {
    const component = create(<MultiQuestionAnswer {...defaultProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
