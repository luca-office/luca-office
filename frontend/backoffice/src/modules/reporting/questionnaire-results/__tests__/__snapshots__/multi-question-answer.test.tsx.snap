// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiQuestionAnswer renders correctly (SC/MC no FreeText) 1`] = `
<div
  className="css-pedd91-MultiQuestionAnswer"
>
  <div
    className="css-1xwuf1o-MultiQuestionAnswer"
  >
    <div
      className="css-13uqss3-MultiQuestionAnswer"
    >
      36 %
    </div>
  </div>
  <div
    className="css-1byc4rd-MultiQuestionAnswer"
  >
    <div
      className="css-13uqss3-MultiQuestionAnswer"
    >
      common__x_of_y
    </div>
    <div
      className="css-17zbm7y-Text"
    >
      test text
    </div>
  </div>
</div>
`;

exports[`MultiQuestionAnswer renders correctly (SC/MC with FreeText) 1`] = `
<div
  className="css-pedd91-MultiQuestionAnswer"
>
  <div
    className="css-1xwuf1o-MultiQuestionAnswer"
  >
    <div
      className="css-13uqss3-MultiQuestionAnswer"
    >
      36 %
    </div>
  </div>
  <div
    className="css-1byc4rd-MultiQuestionAnswer"
  >
    <div
      className="css-13uqss3-MultiQuestionAnswer"
    >
      common__x_of_y
    </div>
    <div
      className="css-17zbm7y-Text"
    >
      test text
    </div>
  </div>
</div>
`;
